lib/rubocop/cop/lint/redundant_safe_navigation.rb

```diff
@@ -154,6 +200,63 @@ def check?(node)
         def condition?(parent, node)
           (parent.conditional? || parent.post_condition_loop?) && parent.condition == node
         end
+
+        # rubocop:disable Metrics
+        def redundant?(node, receiver, checked_nodes)
+          return false unless node
+          return false if checked_nodes.key?(node)
+
+          checked_nodes[node] = true
+
+          case node.type
+          when :def, :class, :module, :sclass
+            return false
+          when :send
+            return !NIL_METHODS.include?(node.method_name) if node.receiver == receiver
+
+            node.arguments.each do |argument|
+              return true if redundant?(argument, receiver, checked_nodes)
+            end
+
+            return true if redundant?(node.receiver, receiver, checked_nodes)
+          when :begin
+            return true if redundant?(node.children.first, receiver, checked_nodes)
+          when :if, :case
+            return true if redundant?(node.condition, receiver, checked_nodes)
+          when :and, :or
+            return true if redundant?(node.lhs, receiver, checked_nodes)
+          when :pair
+            if redundant?(node.key, receiver, checked_nodes) ||
+               redundant?(node.value, receiver, checked_nodes)
+              return true
+            end
+          when :when
+            node.each_condition do |condition|
+              return true if redundant?(condition, receiver, checked_nodes)
+            end
+          when :lvasgn
```

Comment:
`:ivasgn, :cvasgn, :gvasgn, :casgn`?

otherwise stuff like this is a false negative:
```ruby
NAME = user.name
user&.name
```